knora/dsplib/utils/xml_upload.py

```diff
@@ -404,6 +405,9 @@ def get_propvals(
                         for iri_ref in iri_refs:
                             res_id = iri_ref.split(':')[1]
                             iri = resiri_lookup.get(res_id)
+                            if not iri:
+                                raise BaseError(f'Resource cannot be created, because it contains a salsah-Link to '
+                                                f'the following invalid resource: {res_id}.')
```

Comment:
This fixes a bug which appeared during my testing. In some cases, it was possible that `iri` was None, and then line 411 failed, which lead to a difficult stacktrace.